src/control_overlay/mod.rs

```diff
@@ -1,11 +1,96 @@
-pub mod action;
-pub mod overlay;
+use codee::string::FromToStringCodec;
+use leptos::*;
+use leptos_use::storage::use_local_storage;
+use winit::event_loop::EventLoopProxy;
 
-pub use crate::control_state::ControlState;
+use crate::control_overlay::menu::Menu;
+use crate::control_overlay::menu_view::MenuView;
+use crate::messages::{ControlState, IntervalDetails, LabEvent};
 
-// export enum GlobalMode {
-//     Sphere = "sphere",
-//     Mobius = "mobius",
-//     Klein = "klein",
-//     Construction = "construction",
-// }
+pub mod menu;
+mod menu_view;
+
+#[component]
+pub fn ControlOverlayApp(
+    menu: ReadSignal<Menu>,
+    control_state: ReadSignal<ControlState>,
+    set_control_state: WriteSignal<ControlState>,
+    event_loop_proxy: EventLoopProxy<LabEvent>,
+) -> impl IntoView {
+    let (menu_choice, set_menu_choice) = create_signal(menu.get_untracked().root().clone());
```

Comment:
```suggestion
    let menu_choice = move || menu.get().root().clone();
```

set_menu_choice?????